irc/socket.go

```diff
@@ -94,16 +100,41 @@ func (socket *Socket) Read() (string, error) {
 		return "", io.EOF
 	}
 
-	lineBytes, err := socket.reader.ReadBytes('\n')
+	var err error
 
-	// convert bytes to string
-	line := string(lineBytes[:])
+	for {
+		var newByte byte
+		newByte, err = socket.reader.ReadByte()
```

Comment:
If I understand correctly, these library functions do what we want:

https://golang.org/pkg/bufio/#NewReaderSize
https://golang.org/pkg/bufio/#Reader.ReadLine

i.e., we can set up a `bufio.Reader` that has a buffer at least as big as `maxReadQBytes`, then call `ReadLine` on it; if `isPrefix` is true, then the readq size was exceeded and they get the boot. (Unfortunately, `ReadLine` shaves off the actual delimiters --- will that play nice with `ircmsg`?)

Reply:
(Also, the default size is 4096 bytes --- is that bigger than any client is allowed to negotiate? If so, we could just explicitly hardcode the size at 4096.)

Reply:
Yeah we can certainly do something that's better than this for sure. If there's something that basically lets us say "either a max of X bytes or when we receive a newline" then that'd be optimal (looks like that's what those lib functions do).

With the size, server owners can make those limits pretty much as high as they want. We'll prolly wanna ensure they're `uint64`s (to match `MaxSendQBytes`). However, the value won't change after the server goes up (can't change line len limits with a rehash), so we could just store it as a var on `Server` and return that with a getter, `server.MaxSocketRecvQ()` or something?

Reply:
Yeah, legit.